src/lib/parser/statement/splitter.rs

```diff
@@ -109,19 +120,40 @@ impl<'a> StatementSplitter {
             math_paren_level: 0,
         }
     }
+
+    fn get_statement(&mut self, character: u8) -> StatementVariant<'a> {
+        let variant = if self.flags.contains(Flags::AND) {
+                self.flags.remove(Flags::AND);
```

Comment:
self.flags = self.flags - Flags::AND + new_flag;

Reply:
will do